### PR TITLE
Add snapshot device that draws beam centre

### LINKED ISSUE #471: Draw Beam Centre on snapshot
For diagnostic purposes it is useful to have an idea of where the beam center is when taking a snapshot. For an initial first pass it is ok to just have a cross for the centre, rather than the full shape of the beam that GDA has.

## Acceptance Criteria
* It is possible to take a snapshot and the OAV will put the beam crosshairs on the image
* Behavior remains unchanged for gridscan snapshots

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #471 

### Instructions to reviewer on how to test:
1. Confirm new test passes
2. Run:
```python
import os

os.environ["BEAMLINE"] = "i03"
from dodal.beamlines import i03

oav = i03.oav()

oav.parameters.beam_centre_i = 40
oav.parameters.beam_centre_j = 40


oav.snapshot.directory.set("/tmp/")
oav.snapshot.filename.set("test_pic")
oav.snapshot.trigger()
```
and confirm that `/tmp/test_pic_with_crosshair.png` looks sensible and has a crosshair.

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

**Review by olliesilvester — APPROVED** [review]
A few optional suggestions, but looks good. The test you have in the PR description produces two images: `test_pic.png` (no crosshair) and `test_pic_with_crosshair.png`, I guess two are produced because `snapshot` and `grid_snapshot` are the same PV. If that's intended behaviour, looks good to merge

File: src/dodal/devices/areadetector/plugins/MJPG.py [hunk 1]
```diff
@@ -45,3 +58,23 @@ def get_snapshot():
 
     def post_processing(self, image: Image.Image):
         pass
+
+
+class SnapshotWithBeamCentre(MJPG):
+    CROSSHAIR_LENGTH = 20
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/areadetector/plugins/MJPG.py:64]: Have updated to include units, the number is just one I came up with and looked fine. I will ask the scientist for comment

File: src/dodal/devices/oav/oav_detector.py [hunk 2]
```diff
@@ -82,152 +71,34 @@ def set(self, level_to_set: str) -> StatusBase:
...
 class OAV(AreaDetector):
     cam = ADC(CamBase, "-DI-OAV-01:CAM:")
     roi = ADC(ROIPlugin, "-DI-OAV-01:ROI:")
     proc = ADC(ProcessPlugin, "-DI-OAV-01:PROC:")
     over = ADC(OverlayPlugin, "-DI-OAV-01:OVER:")
     tiff = ADC(OverlayPlugin, "-DI-OAV-01:TIFF:")
     hdf5 = ADC(HDF5Plugin, "-DI-OAV-01:HDF5:")
-    snapshot = Component(SnapshotWithGrid, "-DI-OAV-01:MJPG:")
+    grid_snapshot = Component(SnapshotWithGrid, "-DI-OAV-01:MJPG:")
+    snapshot = Component(SnapshotWithBeamCentre, "-DI-OAV-01:MJPG:")
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/oav/oav_detector.py:82]: I thought about this. I think for most cases people will want the beam centre on there so leaving the snapshot as defaulting to this makes sense I think.

**Comment by DominicOram** [comment]
> The test you have in the PR description produces two images: test_pic.png (no crosshair) and test_pic_with_crosshair.png, I guess two are produced because snapshot and grid_snapshot have the same PV. If that's intended behaviour, looks good to merge

Yh, that was the easiest behavior given what currently existed but on reflection I think it's wrong. I'll change it

File: src/dodal/devices/areadetector/plugins/MJPG.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/areadetector/plugins/MJPG.py]: Neil is happy with size but would like blue rather than white

**Review by olliesilvester — CHANGES_REQUESTED** [review]
This seems nicer, although it has caused some Hyperion tests to break for me locally (in `test_grid_detection_plan.py`). Can you fix these?

**Pull request closed without merging**